assets/app/scripts/directives/logViewer.js

```diff
@@ -82,13 +97,68 @@ angular.module('openshiftConsole')
                 });
               }
             };
-            $win.scroll(onScroll);
 
 
-            var onResize = _.debounce(updateScrollLinks, 50);
+
+            var attachScrollEvents = function() {
+              if(window.innerWidth < BREAKPOINTS.screenSmMin) {
+                $win.on('scroll', onScroll);
+                $cachedScrollableNode.off('scroll', onScroll);
+              } else {
+                $win.off('scroll', onScroll);
+                $cachedScrollableNode.on('scroll', onScroll);
+              }
+            };
+
+
+            // the class .target-logger-node is needed to adjust some
+            // css when the target is not the window.
+            // TODO: resize event breaks the affix, even with this if/else.
+            // however, on first load of either mobile or non this works fine.
+            var affix = function() {
+              if(window.innerWidth < BREAKPOINTS.screenSmMin) {
+                $affixableNode
+                  .removeClass('target-logger-node')
+                  .affix({
+                    target:  window,
+                    offset: {
+                        top:  $scope.followAffixTop || 0, // 390,
+                        bottom: $scope.followAffixBottom || 0 // 90
+                    }
+                  });
+              } else {
+                $affixableNode
+                  .addClass('target-logger-node')
+                  .affix({
+                    target:  $cachedScrollableNode,
+                    offset: {
+                        top: $scope.followAffixTop || 0, // 390,
+                        bottom: $scope.followAffixBottom || 0 // 90
+                    }
+                  });
+              }
+            };
```

Comment:
Have you tried adding `$affixableNode.affix('checkPosition')`?

> This method needs to be called whenever the dimensions of the affixed content or the target element are changed, to ensure correct positioning of the affixed content.

http://getbootstrap.com/javascript/#affixcheckposition